javarosa/src/test/java/org/javarosa/form/api/test/FormatDateTest.java

```diff
@@ -0,0 +1,75 @@
+package org.javarosa.form.api.test;
+
+import org.javarosa.core.model.FormIndex;
+import org.javarosa.core.model.QuestionDef;
+import org.javarosa.core.model.data.DateData;
+import org.javarosa.core.model.data.IntegerData;
+import org.javarosa.core.services.locale.Localization;
+import org.javarosa.core.services.locale.Localizer;
+import org.javarosa.core.test.FormParseInit;
+import org.javarosa.form.api.FormEntryController;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * @author wspride
+ */
+public class FormatDateTest {
+    private FormParseInit fpi;
+
+    /**
+     * load and parse form
+     */
+    @Before
+    public void initForm() {
+        System.out.println("init FormDateTest");
+        fpi = new FormParseInit("/format_date_tests.xml");
+    }
+
+    /**
+     * Tests whether format-date works for date strings,
+     * both wrapped and not wrapped by date()
+     */
+    @Test
+    public void testAnswerQuestion() {
+        DateData ans;
+        FormEntryController fec = fpi.getFormEntryController();
+        FormEntryPrompt prompt;
+        fec.jumpToIndex(FormIndex.createBeginningOfFormIndex());
+        Localizer l = fpi.getFormDef().getLocalizer();
+        l.setDefaultLocale(l.getAvailableLocales()[0]);
+        l.setLocale(l.getAvailableLocales()[0]);
+
+        do {
```

Comment:
do/while/continue isn't needed if you add one additional `fec.stepToNextEvent()` call